src/py/mat3ra/made/tools/build/interface.py

```diff
@@ -1,268 +1,206 @@
-        self, termination: Union[int, TerminationType], strain_mode: StrainModes = StrainModes.mean_abs_strain
-    ) -> List[Interface]:
-        return sorted(
-            self.get_interfaces_for_termination_sorted_by_strain(termination, strain_mode),
-            key=lambda x: x.num_sites,
+    _BuildParametersType = ZSLStrainMatchingInterfaceBuilderParameters
+    _GeneratedItemType = PymatgenInterface
+
+    def _generate(self, configuration: InterfaceConfiguration) -> List[PymatgenInterface]:
+        generator = ZSLGenerator(**self.build_parameters.strain_matching_parameters.dict())
+        builder = CoherentInterfaceBuilder(
+            substrate_structure=to_pymatgen(configuration.substrate_configuration.bulk),
+            film_structure=to_pymatgen(configuration.film_configuration.bulk),
+            substrate_miller=configuration.substrate_configuration.miller_indices,
+            film_miller=configuration.film_configuration.miller_indices,
+            zslgen=generator,
         )
 
-    def sort_interfaces_for_all_terminations_by_strain_and_size(self):
-        for termination in self.terminations:
-            self.set_interfaces_for_termination(
-                termination, self.get_interfaces_for_termination_sorted_by_strain_and_size(termination)
-            )
-
-    def get_all_interfaces(self) -> List[Interface]:
-        return functools.reduce(lambda a, b: a + b, self.data.values())
-
-    def get_interfaces_as_materials(
-        self, termination: Union[int, TerminationType], slice_range_or_index: Union[int, slice]
-    ) -> List[Interface]:
-        return list(
-            map(
-                convert_atoms_or_structure_to_material,
-                self.get_interfaces_for_termination(termination, slice_range_or_index),
-            )
+        interfaces = builder.get_interfaces(
+            termination=configuration.termination_pair,
+            gap=configuration.distance_z,
+            film_thickness=configuration.film_configuration.thickness,
+            substrate_thickness=configuration.substrate_configuration.thickness,
+            in_layers=True,
         )
+
+        return list([interface_patch_with_mean_abs_strain(interface) for interface in interfaces])
+
+    def _sort(self, items: List[_GeneratedItemType]):
+        # TODO: sort by number of atoms
+        return sorted(items, key=lambda x: np.mean(np.abs(x.interface_properties[StrainModes.mean_abs_strain])))
+
+    def _post_process(self, items: List[_GeneratedItemType], post_process_parameters=None) -> List[Material]:
+        materials = [Material(from_pymatgen(interface)) for interface in items]
+        strains = [interface.interface_properties[StrainModes.mean_abs_strain] for interface in items]
+
+        for material, strain in zip(materials, strains):
+            material.metadata = {"mean_abs_strain": strain}
+            material.name = f'{material.name}, Interface, Strain:{material.metadata["mean_abs_strain"] * 100:.3f}%'
```

Comment:
Did we want to add miller indices to the name?

Let's isolate the name modification to a separate function - can be inside the parent class